### ISSUE TITLE
hz.currentUser doesn't execute until something else is executed

### ISSUE THREAD

**Issue by steelbrain** [rethinkdb/horizon#567]
- Server version v1.1.1
- Client version The same I think

``` js
const hz = Horizon();
hz.currentUser().fetch().subscribe( (user) => console.log(JSON.stringify(user)) );
```

^ this example does not work unless you perform another query or explicitly do `hz.connect()`

**Comment by deontologician** [comment]
Also, as part of fixing this, I'll expose the currentUserId as well

**Comment by deontologician** [comment]
Reopening since @cybor0 on the forums has run into this again: https://discuss.horizon.io/t/index-html-and-currentuser-problem/288/1

**Comment by deontologician** [comment]
This is more general, no read query is emitted until `.subscribe` is called on the socket (which is what `.connect()` does). So this needs to be fixed still

**Comment by deontologician** [comment]
Fix is up in #653
